File: usecases/src/main/java/com/zeyad/usecases/exceptions/NetworkConnectionException.java
package com.zeyad.usecases.exceptions;

/**
 * Exception throw by the application when a there is a network connection exception.
 */
public class NetworkConnectionException extends Exception implements IErrorBundle {

    public NetworkConnectionException() {
        super();
    }

    public NetworkConnectionException(final String message) {
        super(message);
    }

    public NetworkConnectionException(final String message, final Throwable cause) {
        super(message, cause);
    }

    public NetworkConnectionException(final Throwable cause) {
        super(cause);
    }

    @Override
    public Exception getException() {
        return this;
    }
}

File: usecases/src/main/java/com/zeyad/usecases/exceptions/NetworkConnectionException.kt
package com.zeyad.usecases.exceptions

/**
 * Exception throw by the application when a there is a network connection exception.
 */
class NetworkConnectionException(message: String) : Exception(message), IErrorBundle {
    override fun message(): String {
        return localizedMessage
    }

    override fun exception(): Exception {
        return this
    }
}

